app/controllers/course/stage/instructions.ts

```diff
@@ -47,16 +48,32 @@ export default class CourseStageInstructionsController extends Controller {
     return this.model.courseStage.prerequisiteInstructionsMarkdownFor(this.model.activeRepository);
   }
 
+  get shouldHideTestRunnerCardBeforeUserHasSubmitted() {
+    return this.featureFlags.hideTestRunnerCardBeforeUserHasSubmitted;
+  }
+
   get shouldShowFeedbackPrompt() {
     return !this.currentStep.courseStage.isFirst && this.currentStep.status === 'complete';
   }
 
+  get shouldShowLinkToForum() {
```

Comment:
Not sure if this needs to be here, maybe it can be in `FirstStageTutorialCard` directly? I don't see it used anywhere outside.

Reply:
@rohitpaulk, it seems easier to compute `shouldShowLinkToForum` before passing just one prop down, otherwise it can be a mess passing down all the ingredients or pulling all the ingredients from inside `FirstStageTutorialCard` again.

Should I recompute `isCurrentStage` and `currentStep` in FirstStageTutorialCard?

```
  get shouldShowLinkToForum() {
    return (
      this.isCurrentStage &&
      this.currentStep.status !== 'complete' &&
      this.currentStep.testsStatus !== 'passed' &&
      this.shouldHideTestRunnerCardBeforeUserHasSubmitted
    );
  }
```